ui-library-filesharing-chat-composite/api/UploadFileToAzureBlobStore/index.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { AzureFunction, Context, HttpRequest } from '@azure/functions';
+import * as multipart from 'parse-multipart';
+import HTTP_CODES from 'http-status-enum';
+
+const azureStorageConnectionString = process.env['azureStorageConnectionString'];
+let storageAccountName;
+try {
+  console.log("****"+azureStorageConnectionString);
+  storageAccountName = azureStorageConnectionString.match(/AccountName=([^;]+)/)[1];
+} catch {
+  console.warn('Azure storage connection string not provided');
+}
+const fileSharingUploadsContainerName = 'uploads'; // Must be in storage path value in function.json
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const httpTrigger: AzureFunction = async function (context: Context, req: HttpRequest): Promise<any> {
+  context.log('HTTP trigger function processed a request to upload a file to azure blob store.');
+
+  if (!req.body || !req.body.length) {
+    context.res.body = { error: `Request body is not defined` };
+    context.res.status = HTTP_CODES.BAD_REQUEST;
+    return;
+  }
+
+  // `filename` is required property to use multi-part npm package
+  if (!req.query?.filename) {
+    context.res.body = { error: `filename is not defined` };
+    context.res.status = HTTP_CODES.BAD_REQUEST;
+    return;
+  }
+
+  // Content type is required to know how to parse multi-part form
+  if (!req.headers || !req.headers['content-type']) {
+    context.res.body = { error: `Content type is not sent in header 'content-type'` };
+    context.res.status = HTTP_CODES.BAD_REQUEST;
+    return;
+  }
+
+  context.log(
+    `*** Uploading Filename:${req.query?.filename}, Content type:${req.headers['content-type']}, Length:${req.body.length}`
+  );
+
+  if (!azureStorageConnectionString) {
+    throw Error("Storage isn't configured correctly - connection string missing in `azureStorageConnectionString`");
+  }
+
+  try {
+    // Each chunk of the file is delimited by a special string
+    const bodyBuffer = Buffer.from(req.body);
+    const boundary = multipart.getBoundary(req.headers['content-type']);
+    const parts = multipart.Parse(bodyBuffer, boundary);
+
+    // The file buffer is corrupted or incomplete
+    if (!parts?.length) {
+      context.res.body = { error: `File buffer is incorrect`};
+      context.res.status = HTTP_CODES.BAD_REQUEST;
+      return;
+    }
+
+    // filename is a required property of the parse-multipart package
+    if (parts[0]?.filename) console.log(`Original filename = ${parts[0]?.filename}`);
+    if (parts[0]?.type) console.log(`Content type = ${parts[0]?.type}`);
+    if (parts[0]?.data?.length) console.log(`Size = ${parts[0]?.data?.length}`);
+    // Passed to Storage
+    context.bindings.storage = parts[0]?.data;
```

Comment:
what about `parts[1]` etc?

Reply:
@anjulgarg This comment still stands both for this PR and your components one. What about other parts in the multipart request sent to the backend?

Reply:
As per my understanding, other parts are not populated because we only upload 1 file at a time.
Other parts would always be empty for our use case.

Reply:
@prabhjot-msft Let's document that behavior if that's the case :)

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST

This was just my lack of understanding of how a multipart/form-data form works.
Yes, since we're sending only one file, it'll always be the first part. I think it's worth nothing this in a comment here.